src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -419,15 +419,37 @@ public static boolean isReceiverParameter(DetailAST parameterDefAst) {
                 && parameterDefAst.findFirstToken(TokenTypes.IDENT) == null;
     }
 
+    /**
+     * Returns the access modifier of the method/constructor at the specified AST. If
+     * the method is in an interface or annotation block, the access modifier is assumed
+     * to be public.
+     *
+     * @param ast the token of the method/constructor.
+     * @return the access modifier of the method/constructor.
+     */
+    public static AccessModifier getAccessModifierFromModifiersToken(DetailAST ast) {
+        final AccessModifier accessModifier;
+
+        if (ScopeUtil.isInInterfaceOrAnnotationBlock(ast)) {
+            accessModifier = AccessModifier.PUBLIC;
```

Comment:
Since Java 9 there may be [private](https://bugs.openjdk.java.net/browse/JDK-8071453) methods in an interface definition. 
Now the difference between a class and an interface is that the default access modifier for the class is `package` and for the interface is `public`.

Reply:
@pbludov I recommend creating an issue for this. This code is copied from another location and I remember seeing something similar to this in other code.